### PR TITLE
Add 'Debugging Plugin Sources' in  Developing.md

### PR DESCRIPTION
**Pull request by alvsan09**
Added guidelines on how to debug a vscode extension as a
Theia plugin from sources.

Signed-off-by: Alvaro Sanchez-Leon <alvaro.sanchez-leon@ericsson.com>

#### What it does

Adds guidelines on how to debug a vscode extension as a
Theia plugin from sources.

#### How to test

The source can be downloaded and the impacted file opened with an mark down editor for a proper review

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: doc/Developing.md [hunk 1]
```diff
@@ -284,6 +284,67 @@ You should be able to see message of `[${server-name}: ${server-PID}]: IPC start
...
+            "outputCapture": "std"
+        }
+```
+---
+
+#### Producing typescript maps for your plugin
+- tsconfig.json > {} compilerOptions
+```json
+"sourceMap": true
+```
```
[the remark this answers is not in the document]
Comment (alvsan09) [reply doc/Developing.md:326]: Yes, it looks better adding the compilerOpions,
when it comes to new line after header, it's not supported well by renders, I think it loos better with the '-' to make it for an entry of unordered list. 
Let me know what you think !, thanks !

File: doc/Developing.md [hunk 2]
```diff
@@ -284,6 +284,67 @@ You should be able to see message of `[${server-name}: ${server-PID}]: IPC start
...
+If Webpack is used:
+- package.json
+```json
+webpack --mmode development
+```
```
[the remark this answers is not in the document]
Comment (alvsan09) [reply doc/Developing.md]: Addressed in new version of the commit, thanks Paul !

File: doc/Developing.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alvsan09) [reply doc/Developing.md]: I have pushed a new version of the commit, I am marking the header in bold to make a clear separation with the fist line, and as a workaround for the newline issue after headers by the markdown renders.

[the remark this answers is not in the document]
Comment (alvsan09) [reply doc/Developing.md]: I have pushed a new version of the commit, I am marking the header in bold to make a clear separation with the fist line, and as a workaround for the newline issue after headers by the markdown renders.

**Review by marechal-p — APPROVED** [review]
@alvsan09 I forced pushed the last few formatting issues I saw, LGTM now!

**Comment by alvsan09** [comment]
It looks good to me as well, Thanks @marechal-p  !

**Pull request merged**